massa-execution-worker/src/tests/scenarios_mandatories.rs

```diff
@@ -321,6 +321,165 @@ fn send_and_receive_async_message() {
     manager.stop();
 }
 
+/// # Context
+///
+/// Functional test for asynchronous messages sending and handling with a filter
+///
+/// 1. a block is created containing an `execute_sc` operation
+/// 2. this operation deploy a smart contract and call his function `test`
+/// 3. `test` generates an event and place a message to be triggered once again if `test2` datastore key of address `A12DDxjqtBVshdQ4nLqYg6GwRddY5LzEC7bnatVxB5SFtpbCFj8E` is created/modify
+/// 4. we set the created block as finalized so the message is actually sent
+/// 5. we execute the following slots for 300 milliseconds to reach the message execution period
+/// 6. We send a new operation with a smart contract that modify `test` datastore key and so doesn't trigger the message.
+/// 7. We send a new operation with a smart contract that create `test2` datastore key and so trigger the message.
+/// 8. once the execution period is over we stop the execution controller
+/// 9. we retrieve the events emitted by smart contract
+/// 10. `test` handler function should have emitted two events
+/// 11. we check if they are events
+/// 12. if they are some, we verify that the data has the correct value
+#[test]
+#[serial]
+fn send_and_receive_async_message_with_filter() {
```

Comment:
This could use an extra inline comment to mark the line that triggers the SC. Also I believe you left some dead code commented in the test.